kernel/src/kernel.rs

```diff
@@ -638,6 +644,46 @@ impl Kernel {
                         },
                     }
                 }
+                process::State::YieldedFor(upcall_id) => {
+                    // If this process is waiting for a specific upcall, see if
+                    // it is ready. If so, dequeue and execute it, otherwise move on.
```

Comment:
I would use another word instead of *execute*, as this does not actually execute the callback function.

Reply:
"schedule?"

Reply:
Not sure about this, we do not actually schedule the callback, just return its arguments. What about *return its arguments to the application*.